src/platform/plugins/shared/unified_search/public/dataview_picker/data_view_picker.tsx

```diff
@@ -72,6 +72,10 @@ export interface DataViewPickerProps {
    * Makes the picker disabled by disabling the popover trigger
    */
   isDisabled?: boolean;
+  /**
+   * Optional callback when data view picker is closed
+   */
+  onClosePopover?: () => void;
```

Comment:
Can you provide use cases background on where `onClosePopover` would be used?

This seems like an odd use case to have a popover open another popover and then having to pass `onClosePopover` to close these cascading popovers.

Reply:
Just some older design we need to support here I guess

Reply:
@nreese yeah this was added to fix a UI bug. In analyzer, the dataview picker icon persists after opening any flyout

https://github.com/user-attachments/assets/ce74593c-887d-4446-913c-9cf45afc022c

Reply:
Since the data view picker is the only thing displayed in the popup, what about just displaying the change data view popover when users click the original icon button to avoid pop-overs opening other popovers?

Reply:
@nreese Thanks! I will consult the UIUX team on this. 

Currently the button and the popover content are tightly coupled in [change_dataview.tsx](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/unified_search/public/dataview_picker/change_dataview.tsx). To separate them and exposing both options will require a refactor of that file. Is this what you are suggesting?

Reply:
We actually see lots of similar behaviors throughout Kibana, for example when popover are used in flyouts... Having a `onClosePopover` callback isn't a new pattern (I'm not saying it is a good one 😉).

It seems that we would have to do some good amount of refactor in order to get this to work without the `onClosePopover` callback, which increase the risk of breaking existing functionality.

@nreese are you really opposed to adding this callback? Even if it is a temporary solution to get us moving forward? I understand temporary is often a bad word, as things tend to usually be temporary for very long times... but such a callback does seem to me like a lot to maintain right?